scripts/hem.jl

```diff
@@ -0,0 +1,189 @@
+using Revise
+using Gurobi
+using Ipopt
+using JuMP
+using YAML
+
+using HolisticElectricityModel
+import HolisticElectricityModelData
+
+const GUROBI_ENV = Gurobi.Env()
+const HEMDataRepo = HolisticElectricityModelData
+
+# File locations
+base_dir = abspath(joinpath(dirname(Base.find_package("HolisticElectricityModel")), ".."))
+hem_data_dir = dirname(dirname(Base.find_package("HolisticElectricityModelData")))
+input_path = joinpath(hem_data_dir, "inputs")
+test_data_dir = joinpath(base_dir, "test", "driver_outputs")
+
+# Parse inputs
+config = YAML.load_file(joinpath(base_dir, "scripts", "hem_config.yaml"))
+bal_areas = config["balancing_areas"]
+base_year = config["base_year"]
+num_future_years = config["num_future_years"]
+market_structure = config["market_structure"]
+der_use_case = config["der_use_case"]
+supply_choice_use_case = config["supply_choice_use_case"]
+num_ipps = config["num_ipps"]
+ipp_algorithm = config["ipp_algorithm"]
+rate_design = config["rate_design"]
+net_metering_policy = config["net_metering_policy"]
+solver = config["solver"]
+
+# Define scenario
+bal_areas_len = length(bal_areas)
+future_years = [base_year + i for i in range(1, num_future_years)]
+scenario = HEMDataRepo.DataSelection(bal_areas, base_year, future_years, num_ipps)
+
+# need to run in julia: run(output_dir = PROFILES_DIRECTORY, user = "nguo", hostname = HOSTNAME, dbname = DATABASE, port = PORT, pca_ids = nothing) to get residential and commercial profiles
+# also need to run in command prompt: python inputs/write_industrial_profiles.py #ba to get industrial profiles
+
+input_dir_name =
+    "ba_" *
+    "$bal_areas_len" *
+    "_base_" *
+    "$base_year" *
+    "_future_" *
+    "$num_future_years" *
+    "_ipps_" *
+    "$num_ipps" *
+    "_enhanced_test_full"
+input_dir = joinpath(hem_data_dir, "runs", input_dir_name)
+# mkpath(input_dir)
+
+# HEMDataRepo.parse_inputs(input_path, input_dir, scenario)
+
+# Restructure HEM options from user/config file input
+# market structure
+if market_structure == "wholesale_market"
+    market_structure = WholesaleMarket()
+elseif market_structure == "vertically_integrated_utility"
+    market_structure = VerticallyIntegratedUtility()
+else
+    error("Invalid market structure: $market_structure.")
+end
```

Comment:
Instead of if-elseif-else-end statements, should these data be stored in maps/dicts (whatever the right construct is in Julia)? Then this could be one line (apologies for the Python-isms): 

`market_structure = MARKET_STRUCTURES[market_structure]()`